lib/spack/spack/cmd/env.py

```diff
@@ -162,7 +274,15 @@ def env_activate(args):
     if args.env or args.no_env or args.env_dir:
         tty.die("Calling spack env activate with --env, --env-dir and --no-env is ambiguous")
 
-    env_name_or_dir = args.activate_env or args.dir
+    # special parser error handling relative to the --temp flag
+    temp_conflicts = iter([args.keep_relative, args.dir, args.env_name, args.with_view])
+    if args.temp and any(temp_conflicts):
+        tty.die(
+            "spack env activate --temp cannot be combined with managed environments, --with-view,"
+            " --keep-relative, or --dir."
+        )
+
+    env_name_or_dir = args.env_name
```

Comment:
Is `env_name_or_dir` even needed anymore?  If not, you could replace use of `env_name_or_dir` with `args.env_name`.